src/general/FilterArgument.cs

```diff
@@ -0,0 +1,59 @@
+using Godot;
+using System;
+using System.Collections.Generic;
+
+/// <summary>
+///   Parent class for filters arguments
+/// </summary>
+/// <remarks>
+///   <para>
+///     Used to list arguments of different types. Not generic because dictionaries require non-generic types.
+///   </para>
+/// </remarks>
+public abstract class FilterArgument
+{
+    /// <summary>
+    ///   Helper function to get Number from NumberFilterArgument without cast.
+    /// </summary>
+    public float GetNumberValue()
+    {
+        return (this as NumberFilterArgument)?.Value ??
+            throw new InvalidOperationException("Can't get number value from a non-numeric filter argument!");
+    }
+
+    public string GetStringValue()
+    {
+        return (this as MultipleChoiceFilterArgument)?.Value ??
+            throw new InvalidOperationException("Can't get string value from a non-string filter argument!");
+    }
+
+    public class NumberFilterArgument : FilterArgument
+    {
+        public readonly float MinValue;
+        public readonly float MaxValue;
+
+        public float Value;
+
+        public NumberFilterArgument(float minValue, float maxValue, float defaultValue)
+        {
+            if (defaultValue < minValue || defaultValue > maxValue)
```

Comment:
I think use `Mathf.Clamp()` is better.

Reply:
Isn't Mathf.Clamp() used to simply alter the value? Here, if the default value is not set right it would probably be a user mistake. Which is why I use these checks to throw an exception.

Reply:
I agree that not using clamp here is the right way to check and throw the argument exception.

Reply:
Alright, thanks for the confirmation. I added a comment to explain the logic since there's never too many of them (almost).